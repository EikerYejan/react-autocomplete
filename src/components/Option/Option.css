.option {
  width: 100%;
  font-size: 14px;
  position: relative;
  padding: 6px 15px;
  cursor: pointer;
  text-align: left;
  border-bottom: 1px solid var(--borderColor);
  transition: 0.1s linear;
}

.option:hover,
.option.is--selected {
  background-color: var(--hoverColor);
}
