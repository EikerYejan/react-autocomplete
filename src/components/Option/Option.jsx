import React from 'react'
import PropTypes from 'prop-types'
import './Option.css'

class Option extends React.Component {
  render() {
    const { label, isSelected, searchTerm, ...restProps } = this.props
    const content =
      typeof searchTerm === `string`
        ? label.replace(new RegExp(searchTerm, 'i'), `<span class="option-highlight" >${searchTerm}&nbsp;</span>`)
        : label

    return (
      <li
        {...restProps}
        title={label}
        className={`option ${isSelected && 'is--selected'}`}
        dangerouslySetInnerHTML={{
          __html: content,
        }}
      />
    )
  }
}

Option.propTypes = {
  label: PropTypes.string.isRequired,
  isSelected: PropTypes.bool,
  searchTerm: PropTypes.string,
}

export default Option
