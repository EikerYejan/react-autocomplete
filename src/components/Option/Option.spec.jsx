import { render } from '@testing-library/react'
import React from 'react'
import Option from './Option'

describe('<Option />', () => {
  it('Should render', () => {
    const { baseElement } = render(<Option isSelected label="Label" />)

    expect(baseElement).toBeInTheDocument()
    expect(baseElement).toHaveTextContent('Label')
    expect(baseElement).toMatchSnapshot()
  })
})
