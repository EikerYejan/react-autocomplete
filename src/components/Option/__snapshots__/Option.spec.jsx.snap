// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Option /> Should render 1`] = `
<body>
  <div>
    <li
      class="option is--selected"
      title="Label"
    >
      Label
    </li>
  </div>
</body>
`;
